test/utilities.go

```diff
@@ -0,0 +1,196 @@
+					propValue[1] == "{") {
+				continue
+			}
+			// Any other ommision will not be considered as equal
+			err = fmt.Errorf(line)
+			return false, err
+		}
+		// If the test has properties not found in the baseline, then they are not equal
+		if strings.Contains(line, "___ADDED___") {
+			//			fmt.Printf("XXX j1:\n%s\n", j1)
+			//			fmt.Printf("XXX j2:\n%s\n", j2)
+			err = fmt.Errorf(line)
+			return false, err
+		}
+		// If the properties are different, they they are not equal
+		if strings.Contains(line, "___DIFFER___") {
+			err = fmt.Errorf(line)
+			return false, err
+		}
+	}
+	// Extra layer of check. This is not expected to be true.
+	// If only difference is removed with false value property, then the difference
+	// will be qualified as SupersetMatch. If not, should be be considered equal.
+	if d != jsondiff.SupersetMatch {
+		err = fmt.Errorf(str)
+		return false, err
+	}
+	return true, nil
+}
+
+// SortJsonArrays sorts the elements inside json arrays.
+// Some new line organization may be broken.
+// e.g. ...[  {a: "a3", b: "b3"},   {a: "a1", b: "b1"},    {a: "a2", b: "b2"}]...
+// will be:
+//      ...[{a: "a1", b: "b1"},{a: "a2", b: "b2"},{a: "a3", b: "b3"}]...
+// Note: any characters between [ and {, as well as between elements will be lost: ("...},xxx{..." "xxx" is lost ) 
+// This function will recursively sort arrays inside arrays as well.
+func SortJsonArrays(json string) (sorted string, err error) {
+	idx := 0
+	for idx != len(json) {
+		if json[idx] == '[' {
+			c, err := sortElements(&json, idx+1, len(json), '[', ']')
+			if err != nil {
+				return "", err
+			}
+			idx = c
+		}
+		if json[idx] == '{' {
+			c, err := matchingCloseIndex(&json, idx+1, len(json), '{', '}')
+			if err != nil {
+				return "", err
+			}
+			idx = c
+		}			
+		idx++
+	}
+	return json, nil
+}
+
+// return the index of the character that closes the bracket.
+// If an array is detected, will call sortElements to sort the elements inside the array
+func matchingCloseIndex(json *string, start, end int, ochar, cchar rune) (closeIndex int, err error) {
+
+	for i := start; i < len(*json); i++ {
+		if (*json)[i] == '[' {
+			i, err = sortElements(json, i+1, end, '[', ']')
+			if err != nil {
+				return -1, err
+			}
+			continue
+		}
+		if (*json)[i] == '{' {
+			c, err := matchingCloseIndex(json, i+1, end, ochar, cchar)
+			if err != nil {
+				return -1, err
+			}
+			i = c
+			continue
+		}
+		if (*json)[i] == '}' {
+			// found the match
+			return i, nil
+		}
+		if (*json)[i] == ']' {
+			return -1, fmt.Errorf("Could not find the matching %c", cchar)
+		}
+	}
+	return -1, fmt.Errorf("Could not find the matching %c", cchar)	
+	//	return -1, fmt.Errorf("Could not find the matching %c", cchar)
+}
+
+// sorts the elements inside the array
+// if a nested arrray is detected, it will first sort the elements inside the nested array
+func sortElements(json *string, start, end int, ochar, cchar rune) (closeIndex int, err error) {
```

Comment:
Were you able to verify exactly where the unordered elements came from? We verified with `watch` that Indexer sorts it's response, and the documentation of json/encode claims that it also sorts: https://golang.org/src/encoding/json/encode.go?#L136

Reply:
See my comment above, regarding the default value issue. 
Moreover, both the response and the object jsons do not have proper ordering of the elements inside arrays. This ordering depends on the values, and not just json keys.